src/main/java/com/karankumar/bookproject/backend/service/PredefinedShelfService.java

```diff
@@ -100,15 +120,30 @@ public void populateTestData() {
             populateAuthorRepository();
         }
 
+        if (tagRepository.count() == 0) {
```

Comment:
This could be refactored into a method (e.g. called 'populateTagRepository()') as we have populated the other repositories when the repository count is 0